src/main/java/seedu/address/model/task/TaskManager.java

```diff
@@ -0,0 +1,133 @@
+package seedu.address.model.task;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Comparator;
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.task.exceptions.InvalidSortingOrderException;
+
+/**
+ * Represents a task manager that stores and manages tasks
+ */
+public class TaskManager implements ReadOnlyTaskManager {
+    private static final String COMPARATOR_TYPE_DESCRIPTION = "Description";
+    private static final String COMPARATOR_TYPE_DEADLINE = "Deadline";
+
+    private final TaskList tasks;
+    private Comparator<Task> sortingOrder;
+
+    /**
+     * Constructs a TaskManager object with an empty task list.
+     */
+    public TaskManager() {
+        this.tasks = new TaskList();
+        this.sortingOrder = new Task.TaskDeadlineComparator();
+    }
+
+    /**
+     * Constructs a TaskManager using the tasks in the {@code toBeCopied}
+     */
+    public TaskManager(ReadOnlyTaskManager toBeCopied) {
+        this();
+        requireNonNull(toBeCopied);
+        resetData(toBeCopied);
+    }
```

Comment:
Perhaps there is no need to do a non null check here? Since resetData does a non null check as well.

Reply:
Think it's fine to leave as is for the sake of defensive programming.